pymc_marketing/mmm/base.py

```diff
@@ -472,8 +474,136 @@ def compute_channel_contribution_original_scale(self) -> DataArray:
+    def _plot_estimations(
+        self, x: np.ndarray, ax: plt.Axes, channel: str, color_index: int, x_stop: int
+    ) -> None:
+        """
+        Plot the Michaelis-Menten curve fit for the given channel based on the estimation of the Menten parameters.
+
+        The function computes the mean channel contributions, estimates the Michaelis-Menten parameters, and plots
+        the curve fit. An elbow point on the curve is also highlighted.
+
+        Parameters
+        ----------
+        x : np.ndarray
+            The x-axis data, usually representing the amount of input (e.g., substrate concentration in enzymology terms).
+        ax : plt.Axes
+            The matplotlib axes object where the plot should be drawn.
+        channel : str
+            The name of the channel for which the curve fit is being plotted.
+        color_index : int
+            An index used for color selection to ensure distinct colors for multiple plots.
+
+        Returns
+        -------
+        None
+            The function modifies the given axes object in-place and doesn't return any object.
+        """
+        channel_contributions = self.compute_channel_contribution_original_scale().mean(
+            ["chain", "draw"]
+        )
+
+        L, k = estimate_menten_parameters(channel, self.X, channel_contributions)
+        elbow_y = michaelis_menten(k, L, k)
+
+        if x_stop is not None:
+            x_limit = x_stop
+        else:
+            x_limit = k * (0.99 * L / (L * 0.01))
+
+        x_fit = np.linspace(0, x_limit, 1000)
+        y_fit = michaelis_menten(x_fit, L, k)
+
+        ax.plot(x_fit, y_fit, color=f"C{color_index}", label="Fit Curve", alpha=0.6)
+        ax.plot(
+            k,
+            elbow_y,
+            "go",
+            color=f"C{color_index}",
+            markerfacecolor="white",
+        )
+
+        ax.set(xlabel="Spent", ylabel="Contribution")
+        ax.legend()
+
+    def optimize_channel_budget_for_maximum_contribution(
+        self,
+        total_budget: int,
+        *,
+        parameters: Dict[str, Tuple[float, float]],
+        budget_bounds: Optional[Dict[str, Tuple[float, float]]],
+    ) -> pd.DataFrame:
+        """
+        Optimize the allocation of a given total budget across multiple channels to maximize the expected contribution.
+
+        The optimization is based on the Michaelis-Menten equation, where each channel's contribution
+        follows a saturating function of its allocated budget. The function seeks the budget allocation
+        that maximizes the total expected contribution across all channels.
+
+        Parameters
+        ----------
+        total_budget : int, requiere
+            The total budget to be distributed across channels.
+        parameters : dict, requiere
+            A dictionary where keys are channel names and values are tuples (L, k) representing the
+            Michaelis-Menten parameters for each channel.
+        budget_bounds : dict, optional
```

Comment:
Here do you mean optional as in there is a default value (because there is no at the moment)? Recall that the type hint `Optional` refers to the fact that it can take `None` values. Not that is really optional.

Reply:
Correct, should be optional (None allow). If the user doesn't use it and the value is None then some bounds are used by default.